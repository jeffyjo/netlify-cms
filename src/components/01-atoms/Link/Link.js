import React from 'react'
import Proptypes from 'prop-types'
import { Link as GatsbyLink } from 'gatsby'

import './Link.scss'

const Link = ({ children, to, className }) => {
  return (
    <GatsbyLink className={'a-link ' + (className ? className : '')} to={to}>
      {children}
    </GatsbyLink>
  )
}

Link.propsTypes = {
  linkText: Proptypes.string.isRequired,
  url: Proptypes.string.isRequired,
  className: Proptypes.string,
}

export default Link
