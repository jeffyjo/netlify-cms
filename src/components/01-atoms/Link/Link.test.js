import React from 'react'
import Link from './Link'
import renderer from 'react-test-renderer'
import { render, getByTestId } from 'react-testing-library'

describe('Atom | Link ', () => {
  let component = null
  
  // Json-representation of component
  let tree = null

  beforeEach(() => {
    component = renderer.create(
      <Link className="link-class" url="http://link.com">Link</Link>
      )
      tree = component.toJSON()
    })
    
    test('Link should render', () => {
      expect(tree).toMatchSnapshot()
    })
    
    test('Link should have default CSSclass', () => {
      expect(tree.props.className).toContain('a-link')
    })

    test('Link should be injected with cssCLass', () => {
      expect(tree.props.className).toContain('link-class')
    })
    
    const TestLink = () => <Link url="link">Link</Link>

    test('Link should display the correct link-text', () => {
      const { getByText } = render(<TestLink />)
      expect(getByText("Link")).toHaveTextContent("Link")
    })

    test('Link should have correct link', () => {
      const { getByText } = render(<TestLink />)
      expect(getByText("Link")).toHaveAttribute("href", "link")
    })
})