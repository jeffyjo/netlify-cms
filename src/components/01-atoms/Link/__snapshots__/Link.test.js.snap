// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom | Link  Link should render 1`] = `
<a
  className="a-link link-class"
  href="http://link.com"
>
  Link
</a>
`;
